### PR TITLE
Verify signature dc removal

### PR DESCRIPTION
**Pull request by fabriziogianni7**
# Description

- update encodeRemoveDataCapParameters function to return the txBlob adding the prefix "fil_removedatacap:" at the beginnign of the buffer
- added test in verifySignature.js
- updated version to v2.2.4
- to run test run: `npx jest api`

### CODE REVIEW

File: src/api/api.ts [hunk 1]
```diff
@@ -74,9 +74,18 @@ export class VerifyAPI {
   //   )
   // }
 
-  encodeRemoveDataCapParameters(message: {verifiedClient: string, dataCapAmount: string, removalProposalID: number[]}) {
-    const orderedProposalParams = [SignatureDomainSeparation_RemoveDataCap, message.verifiedClient, message.dataCapAmount, message.removalProposalID]
-    return this.methods.encode(this.methods.RemoveDataCapProposal, orderedProposalParams)
+  encodeRemoveDataCapParameters(message: { verifiedClient: string, dataCapAmount: string, removalProposalID: number[] }) {
+    const orderedProposalParams = [message.verifiedClient, message.dataCapAmount, message.removalProposalID]
+    const prefix = "fil_removedatacap:"
```
Comment (ssallam) [inline src/api/api.ts:79]: please use the constant defined above.

File: src/api/api.ts [hunk 2]
```diff
@@ -74,9 +74,18 @@ export class VerifyAPI {
...
+    const prefix__hex_encoded = Buffer.from(prefix).toString('hex')
+
+    const encoded_params_buffer = this.methods.encode(this.methods.RemoveDataCapProposal, orderedProposalParams)
+    const encoded_hex_params =encoded_params_buffer.toString('hex')
```
Comment (ssallam) [inline src/api/api.ts:83]: check spacing

File: src/samples/api/verifySignature.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    // console.log('\n')
+
+    // const full_buff = prefix.concat(encoded_string_hex)
+    // console.log("full_buff",full_buff)
+    // console.log('\n')
+
+    console.log('66696c5f72656d6f7665646174616361703a83420066460008000000008100')
+    console.log(encoded_object)
+    console.log('\n')
+    console.log(encoded_object === '66696c5f72656d6f7665646174616361703a83420066460008000000008100')
```
Comment (ssallam) [inline src/samples/api/verifySignature.js:39]: as an added check, please compare the bytes array with the one from the verifreg test 
[102 105 108 95 114 101 109 111 118 101 100 97 116 97 99 97 112 58 131 66 0 102 70 0 8 0 0 0 0 129 0]

File: src/api/api.ts (hunk 1 above)
Comment (fabriziogianni7) [reply src/api/api.ts]: done

File: src/api/api.ts (hunk 2 above)
Comment (fabriziogianni7) [reply src/api/api.ts]: done

File: src/samples/api/verifySignature.js (hunk 3 above)
Comment (fabriziogianni7) [reply src/samples/api/verifySignature.js]: done

File: src/api/api.test.ts [hunk 4]
```diff
@@ -40,9 +74,19 @@ describe('should test the api', () => {
...
+
+    const bytes_array_to_compare = [102, 105, 108, 95, 114, 101, 109, 111, 118, 101, 100, 97, 116, 97, 99, 97, 112, 58, 131, 66, 0, 102, 70, 0, 8, 0, 0, 0, 0, 129, 0]
+
+    const hexStringToCompare = '66696c5f72656d6f7665646174616361703a83420066460008000000008100'
+
+    const params = { verifiedClient: 't0102', dataCapAmount: 34359738368, removalProposalID: [0] }
+    const encoded_hex_string = verifierApi.encodeRemoveDataCapParameters(params)
+    expect(encoded_hex_string).toBe(hexStringToCompare)
+    expect(compareStringToBytesArray(encoded_hex_string,bytes_array_to_compare)).toBeTruthy()
+
```
Comment (ssallam) [inline src/api/api.test.ts:85]: space after comma

File: src/samples/api/verifySignature.js (hunk 3 above)
Comment (fabriziogianni7) [reply src/samples/api/verifySignature.js]: run `npx jest api` to run tests

**Review by ssallam — APPROVED** [review]

File: src/samples/api/verifySignature.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  // console.log(pairs)
+
+  const byteArray = []
+  for (let i = 0; i < pairs.length; i++) {
+    const byte = parseInt(pairs[i], 16)
+    byteArray.push(byte)
+  }
+
+  // console.log(byteArray);
+  // console.log(compareArray);
```
[the remark this answers is not in the document]
Comment (fabriziogianni7) [reply src/samples/api/verifySignature.js]: done

File: src/api/api.test.ts (hunk 4 above)
Comment (fabriziogianni7) [reply src/api/api.test.ts:85]: run lint should be ok

**Pull request merged**